baseplate/sidecars/publisher_queue_utils.py

```diff
@@ -0,0 +1,97 @@
+import contextlib
+
+from typing import Optional
+
+import gevent
+
+from baseplate.lib import config
+from baseplate.lib.message_queue import InMemoryMessageQueue
+from baseplate.lib.message_queue import MessageQueue
+from baseplate.lib.message_queue import PosixMessageQueue
+from baseplate.lib.message_queue import RemoteMessageQueue
+from baseplate.lib.message_queue import TimedOutError as MessageQueueTimedOutError
+from baseplate.server import make_listener
+from baseplate.server import make_server
+from baseplate.thrift.message_queue import RemoteMessageQueueService
+from baseplate.thrift.message_queue.ttypes import GetResponse
+from baseplate.thrift.message_queue.ttypes import PutResponse
+from baseplate.thrift.message_queue.ttypes import TimedOutError as ThriftTimedOutError
+
+
+class RemoteMessageQueueHandler:  # On the queue server, create the queue and define get/put using the InMemoryQueue implementation
+    def is_healthy(self) -> bool:
+        pass
+
+    def __init__(self):
+        self.queue_list = {}
+
+    def get(
+        self, queue_name: str, max_messages: int, timeout: Optional[float] = None
+    ) -> GetResponse:
+        try:
+            # Create queue if doesnt exist
+            # We need to create the queue on both get & put - if get() is called on a queue before it exists, we
+            # still want to wait the appropriate timeout in case anyone puts elements in it
+            queue = self.queue_list.get(queue_name)
+            if not queue:
+                queue = InMemoryMessageQueue(queue_name, max_messages)
+                self.queue_list[queue_name] = queue
+            # Get element from list, waiting if necessary
+            result = queue.get(timeout)
+        except MessageQueueTimedOutError:
+            raise ThriftTimedOutError()
```

Comment:
Just wondering what the reason is for changing the error type here? Also you are throwing away any error info by doing this. It is better to `raise from` the old error so context is passed on:
```python
except MessageQueueTimedOutError as mqtoe:
    raise ThriftTimedOutError from mqtoe
```

Reply:
The thrift server only expects to raise its own thrift exception types - otherwise it raises a generic `thrift.Thrift.TApplicationException: Internal error`. So I'm basically taking our message_queue exception from the InMemoryQueue, translating it to a Thrift exception, then translating it back to a message_queue exception on the client side... I would love a better way to do this, it feels yucky